pkg/httpclient/client.go

```diff
@@ -0,0 +1,202 @@
+// Package httpclient is a simple HTTP client which supports sending and receiving JSON strings using
+// GET, POST, PUT, PATCH, and DELETE requests, with configurable timeouts.
+//
+// To create a new client, you have to call the following code:
+//
+// 		client := httpclient.NewClient()
+//
+// If you want to adjust the timeouts:
+//
+//		timeouts := InitTimeouts()
+//		// adjust any timeouts here
+//		client := httpclient.NewClientWithTimeouts(timeouts)
+//
+// Then, to make a request, call one of the service methods, e.g.:
+//		resp := client.GetJSON("http://site/path")
+//
+// Once you have an user and a corresponding public API key, you can issue authenticated requests,
+// by constructing a new client with the appropriate credentials:
+//
+//		client := httpclient.NewClientWithAuthentication(username, publicAPIToken)
+//
+package httpclient
+
+import (
+	"fmt"
+	"github.com/mongodb-labs/pcgc/pkg/useful"
+	"gopkg.in/errgo.v1"
+	"io"
+	"net"
+	"net/http"
+	"runtime"
+
+	digest "github.com/Sectorbob/mlab-ns2/gae/ns/digest"
```

Comment:
you can omit the `digest` part, also I don't know why `fmt` fails now on this file but imports should be ordered

Reply:
are you running `make fmt`?

Reply:
yes, or even from intellij it won't flag it or fix it  🤷‍♂